test/discriminating_facets_test.py

```diff
@@ -0,0 +1,89 @@
+import unittest
+from mlapi.model.facet import Facet
+from mlapi.question_generator import DiscriminatingFacetsAlgo
+import statistics
+
+class TestDiscriminatingFacets(unittest.TestCase):
+    def test_when_no_discriminating_facet_then_return_no_facet(self):
+        discriminating_algo = DiscriminatingFacetsAlgo()
+        facet_a1 = Facet('NameA', 'ValueA1')
+        facet_a2 = Facet('NameA', 'ValueA2')
+        facet_b = Facet('NameB', 'ValueB')
+        facets_by_document = {'1': [facet_a1], '2': [facet_a2], '3': [facet_b]}
+
+        discriminating_facets = discriminating_algo.get_discriminating_facets(facets_by_document)
+        self.assertEqual(len(discriminating_facets), 0)
+
+    def test_when_1_discriminating_facet_then_return_1_facet(self):
+        discriminating_algo = DiscriminatingFacetsAlgo()
+        facet_a1 = Facet('NameA', 'ValueA1')
+        facet_a2 = Facet('NameA', 'ValueA2')
+        facet_b = Facet('NameB', 'ValueB')
+        facets_by_document = {'1': [facet_a1], '2': [facet_a2], '3': [facet_a2], '4': [facet_b]}
+
+        discriminating_facets = discriminating_algo.get_discriminating_facets(facets_by_document)
+        self.assertEqual(len(discriminating_facets), 1)
+        self.assertEqual(discriminating_facets["NameA"], ['ValueA1', 'ValueA2'])
```

Comment:
consistent use of single/double quote (' vs "). I don't mind which one, but we should stick with it (see what's used in the rest of the project)
```suggestion
        self.assertEqual(discriminating_facets['NameA'], ['ValueA1', 'ValueA2'])
```